core-ui/src/hooks/useTracking.js

```diff
@@ -1,53 +1,56 @@
 import { useEffect } from 'react';
+import LuigiClient from '@luigi-project/client';
 import { useLocation } from 'react-router-dom';
-import { baseUrl } from 'shared/hooks/BackendAPI/config';
-import { useConfig } from 'shared/contexts/ConfigContext';
-import { useFeature } from 'shared/hooks/useFeature';
-import { v4 as uuid } from 'uuid';
+import { useMicrofrontendContext } from 'shared/contexts/MicrofrontendContext';
 
-function getLoggingId() {
-  const STORAGE_KEY = 'busola.logging-id';
-
-  let loggingId = localStorage.getItem(STORAGE_KEY);
-  if (!loggingId) {
-    loggingId = uuid();
-    localStorage.setItem(STORAGE_KEY, loggingId);
-  }
-  return loggingId;
+function sendTrackingRequest(body) {
+  LuigiClient.sendCustomMessage({ id: 'busola.tracking', body });
 }
 
-export function useTracking() {
-  const { isEnabled } = useFeature('TRACKING');
-
+export function usePageViewTracking() {
   const { pathname } = useLocation();
-  const { fromConfig } = useConfig();
 
   useEffect(() => {
-    if (!isEnabled) return;
-
-    let path;
+    let navigationPath;
     if (pathname.includes('/namespaces/')) {
       if (new RegExp('/namespaces/[a-z0-9-]+/?(details)?$').test(pathname)) {
         // namespace details
-        path = 'namespaces';
+        navigationPath = 'namespaces/';
       } else {
         // other resource details
-        path = pathname.replace(new RegExp('/namespaces/.*?/'), '');
+        navigationPath = pathname.replace(new RegExp('/namespaces/.*?/'), '');
       }
     } else {
       // no namespace
-      path = pathname.substring(1);
+      navigationPath = pathname.substring(1);
     }
-    const pathSegments = path.split('/'); // split by '/', take only first part
-    let log;
-    if (pathSegments.length > 1) {
-      log = 'DETAILS ' + pathSegments[0];
-    } else {
-      log = 'LIST ' + pathSegments[0];
+    const pathSegments = navigationPath.split('/'); // split by '/', take only first part
+    let viewType = pathSegments.length > 1 ? 'DETAILS' : 'LIST';
+    let path = pathSegments[0];
+
+    if (pathname === '/overview') {
+      path = 'clusters';
+      viewType = 'DETAILS';
     }
-    fetch(baseUrl(fromConfig) + '/tracking', {
-      method: 'POST',
-      body: JSON.stringify({ type: 'PATH', payload: log, id: getLoggingId() }),
-    }).catch(e => console.debug('Tracking call failed', e));
-  }, [fromConfig, pathname, isEnabled]);
+
+    sendTrackingRequest({
+      event: 'PAGE_VIEW',
+      data: { path, viewType },
+    });
+  }, [pathname]);
+}
+
+export function useSessionStartTracking() {
+  const { cluster } = useMicrofrontendContext();
+
+  useEffect(() => {
+    sendTrackingRequest({
+      event: 'SESSION_START',
+      data: {
+        hostname: window.location.hostname,
+        apiServerAddress: cluster?.cluster.server || null,
+      },
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I'm not sure about the intention. If this should change when a user switches a cluster, then maybe add `cluster?.cluster.server` to the dep array. If this fires only once per the app life, they maybe use the `useComponentDidMount` hook?

Reply:
This thing should fire only once, when the app is loaded - so it's fine, but I probably merge it with `trackClusterChange` from `core`.